File: 650-2-keys-keyboard/2-keys-keyboard.kt
class Solution {
    fun minSteps(n: Int): Int {
        if (n == 1) return 0

        var steps : Int = 0
        var factor : Int = 2
        var nV : Int = n

        while (nV > 1) {
            while (nV % factor == 0) {
                steps += factor
                nV /= factor
            }
            factor++
        }
        return steps
    }
}

File: 650-2-keys-keyboard/2-keys-keyboard.php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function minSteps($n) {
        if ($n == 1) return 0;

        $steps = 0;
        $factor = 2;

        while ($n > 1) {
            while ($n % $factor == 0) {
                $steps += $factor;
                $n /= $factor;
            }
            $factor++;
        }
        return $steps;
    }
}